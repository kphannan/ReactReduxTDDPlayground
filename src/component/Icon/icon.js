
import React, { Component } from 'react';
import 'uxicon/uxicon.css';

export default class Icon extends Component {

  render() {
    return (
      <span style={ this.props.styles } className={`icon_${ this.props.name } ${ this.props.color }-text`}></span>
    );
  }
}

