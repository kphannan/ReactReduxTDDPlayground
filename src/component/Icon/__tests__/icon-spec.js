import React from 'react'
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';
import Icon from '../icon';

Enzyme.configure({ adapter: new Adapter() })

describe('Icon Testing', () => {

  const component = mount(<Icon name="information_outlined" color="orange" styles={{ paddingRight : 10 }}/>)

  test('verify against icon snapshot', () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('verify icon renders on page', () => {
    expect(component.containsAllMatchingElements([
      <Icon />
    ])).toBeDefined()
  })

  test('verify icon css matches UX style guide title class', () => {
    expect(component.find('.icon_information_outlined').length).toBe(1);
    expect(component.find('.orange-text').length).toBe(1);
    expect(component.get(0).props.styles.paddingRight).toBe(10);
  });

})