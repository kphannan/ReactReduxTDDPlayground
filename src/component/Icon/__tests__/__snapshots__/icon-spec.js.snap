// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Testing verify against icon snapshot 1`] = `
<Icon
  color="orange"
  name="information_outlined"
  styles={
    Object {
      "paddingRight": 10,
    }
  }
>
  <span
    className="icon_information_outlined orange-text"
    style={
      Object {
        "paddingRight": 10,
      }
    }
  />
</Icon>
`;
